### PR TITLE
feat: add OrderDerails page with mocked data, add DataGrid dependency

### PR DESCRIPTION
**Pull request by kamilszalast**

### CODE REVIEW

**Comment by kamilszalast** [comment]
I add OrderDerails page, with mocked data. If everything is ok, I try to get data prom API

File: src/pages/OrderDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import {
+  OrderDetail,
+  Product,
+  PaymentStatus,
+  OrderObject,
+} from "../types/table";
+import { getProducts } from "../components/lib/products";
+import { DataGrid } from "@mui/x-data-grid";
```
Comment (dblach-intive) [inline src/pages/OrderDetails.tsx:10]: We can use our Table component

File: src/pages/OrderDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  { id: 3, orderId: 2, productId: 102, quantity: 4 },
+  { id: 4, orderId: 3, productId: 184, quantity: 7 },
+  { id: 5, orderId: 2, productId: 103, quantity: 1 },
+  { id: 6, orderId: 1, productId: 102, quantity: 11 },
+  { id: 7, orderId: 3, productId: 102, quantity: 2 },
+  { id: 12, orderId: 3, productId: 99, quantity: 2 },
+];
+//The same as above
+
+const setOrderDetailRows = (
```
Comment (dblach-intive) [inline src/pages/OrderDetails.tsx:66]: Create separate row for order just like we used it everywhere

File: src/pages/OrderDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      <DataGrid
+        sx={{
+          ".light-blue--header": {
+            backgroundColor: "rgba(116,182,247,0.4)",
+          },
+        }}
+        autoHeight
+        disableSelectionOnClick
+        disableColumnMenu
+        columns={[
```
Comment (dblach-intive) [inline src/pages/OrderDetails.tsx:142]: use our Table component and put these kind of data as a props from parent

File: package.json [hunk 4]
```diff
@@ -12,6 +12,7 @@
     "@material-ui/icons": "^4.11.2",
     "@mui/icons-material": "^5.4.2",
     "@mui/material": "^5.4.2",
+    "@mui/x-data-grid": "^5.8.0",
```
Comment (dblach-intive) [inline package.json:15]: we have table component

File: src/pages/OrderDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        setLoading(false);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, []);
+
+  if (isLoading) return <div>Loading ...</div>;
+
+  const detailRows = setOrderDetailRows(mockedOrderDetails, apiProducts, id);
```
Comment (dblach-intive) [inline src/pages/OrderDetails.tsx:115]: u can use useMemo, btw if we're using parsers like this u can just create utils/index.ts and there u can keep any parsers or helpers setOrderDetailRows

**Review by dblach-intive — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (dblach-intive) [inline package.json:15]: we should remove this one

Comment (kamilszalast) [reply package.json]: removed

**Pull request merged**